### PR TITLE
python-api: improvements to bidirectional sessions

### PR DESCRIPTION
**Pull request by mlyle**
* Properly add synchronization / a defined location where things are written.
* Fix a nasty bug in HID that resulted in duplicated writes and incorrect prefixes
* Teach telemetry API how to handle nacks and acks and ackreqs
* Add a simple program to show off editing and saving an object

### CODE REVIEW

**Comment by mlyle** [comment]
Tested-- partial matrix of python2/3, dronin-logview, dronin-dumplog, dronin-halt, dronin-savetest, dronin-getconfig across USB & with files where appropriate.

File: python/dronin-getconfig [hunk 1]
```diff
@@ -26,36 +29,47 @@ def main():
 
     settings_objects = tStream.uavo_defs.get_settings_objects()
 
-    pipeline = []
+    pipeline = set()
```
Comment (tracernz) [inline python/dronin-getconfig]: This doesn't appear anywhere in the code?

Comment (mlyle) [reply python/dronin-getconfig]: I guess it doesn't anymore :)

File: python/dronin-getconfig [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env python
 
+from __future__ import print_function
```
Comment (tracernz) [inline python/dronin-getconfig:3]: :+1:

**Review by tracernz — APPROVED** [review]

**Comment by mlyle** [comment]
jenkins, test this please

**Pull request merged**